tests/test_engine.py

```diff
@@ -52,6 +52,18 @@ def test_get_lookup_url(self):
         assert final_uri_with_params == '1/2/?extra_param=3'
         SampleResourceModel._lookup_urls = []
 
```

Comment:
This is a good suggestion and I think it may be necessary to make the parameterize use below actually test different scenarios. Insertion order only indirectly affects iteration order of built-in dict, so the parameterize below does not really distinguish between as many different orderings as it looks like it does. Check out how the following outputs do not correspond when you run them:

```
import itertools as it
[xs for xs in it.permutations("abAB")]
[list({x: '' for x in xs}) for xs in it.permutations("abAB")]
```